src/server/nuclearnet/fake_nuclearnet_server.ts

```diff
@@ -84,6 +84,9 @@ export class FakeNUClearNetServer {
 export function hashType(type: string): Buffer {
   // Matches hashing implementation from NUClearNet
   // See https://goo.gl/6NDPo2
-  const hashString: string = XXH.h64(type, 0x4e55436c).toString(16)
+  let hashString: string = XXH.h64(type, 0x4e55436c).toString(16)
+  // The hash string may truncate if it's smaller than 16 characters so we pad it with 0s
+  hashString = `0000000000000000${hashString}`.slice(-16)
```

Comment:
How about `('0'.repeat(16) + hashString).slice(-16)`, so you can see both need to be 16?